### PR TITLE
Upgrade jQuery from 1.x to 3.x

### PR DESCRIPTION
**Pull request by viclim**
Fix https://github.com/GSA/search-gov/issues/2

The following pages are now using jQuery3:
- `/admin/*`
- `/search`
- `/sites/:id`
- Any sites importing `assets/sayt_loader.js` will render the search bar using jQuery 3 and jQuery UI 1.12

`/public/javascripts/jquery/jquery.min.js` has been updated from `1.7.1` to `3.3.1`. Sites referring to it might experience some problems.

### CODE REVIEW

**Comment by viclim** [comment]
Fixing the failing cuke test.

**Comment by viclim** [comment]
@noremmie @nickmarden 
Build for this PR is failing due to the missing ENV CC_TEST_REPORTER_ID.
Is there something I'm missing here?

File: public/javascripts/jquery/jquery-1.7.1.min.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/*! jQuery v1.7.1 jquery.com | jquery.org/license */
```
Comment (nickmarden) [inline public/javascripts/jquery/jquery-1.7.1.min.js]: I'm confused why this branch would have a file called `jquery-1.7.1.min.js` in it. Aren't we trying to get _rid_ of jQuery 1.x?

File: Gemfile [hunk 2]
```diff
@@ -99,7 +99,7 @@ gem 'less-rails-bootstrap', '~> 3.3.5'
 gem 'compass-rails', '~> 2.0.5'
 gem 'compass-blueprint', '~> 1.0.0'
 gem 'jquery-ui-rails', '~> 6.0.1'
-gem 'jquery-rails', '~> 3.1.4'
+gem 'jquery-rails', git: 'https://github.com/viclim/jquery-rails', ref: '3d9985112690c68552a8a10abfdb5ea80b40dec9'
```
Comment (nickmarden) [inline Gemfile]: vimlim/jquery-rails branches from indirect/jquery-rails, not rails/jquery-rails...any reason why?

Comment (viclim) [reply Gemfile]: @nickmarden I forked it really long time ago from there. Let me see if I can correct it.

**Comment by nickmarden** [comment]
@viclim the reason for the missing CC_TEST_REPORTER_ID is that we had previously disabled "pass secrets to forked build requests" in CircleCI. Since [this commit](https://github.com/GSA/search-gov/commit/09ae4aedd0d313244dd66014c59d0115169f5cb7), we have no secrets in our CircleCI except for the CC_TEST_REPORTER_ID. Therefore I have enabled secret passing and restarted the build.

I suppose people are welcome to use that "secret" if they want to report spurious results to the CodeClimate servers about the project.

File: Gemfile (hunk 2 above)
Comment (viclim) [reply Gemfile]: Okay. It's now forked from rails/jquery-rails.

**Comment by viclim** [comment]
@nickmarden 
Is there any affiliate still using the non-mobile search page?
I can't find a production site that uses it to test this out. https://github.com/GSA/search-gov/blob/master/app/assets/javascripts/legacy/application.js#L181-L195

**Comment by dawnpm** [comment]
@viclim Hi - There are several affiliates still using the non-responsive serp, but none of them are using our type-ahead search suggestion script.

**Comment by nickmarden** [comment]
@viclim try

**Comment by nickmarden** [comment]
@viclim try

treas
housingandurbandevelopment

**Comment by nickmarden** [comment]
@viclim could you look at the CodeClimate complaint?

**Comment by viclim** [comment]
@nickmarden the codeclimate complaint is from the legacy js. Somewhat related to the question I posted about the non-mobile searches page. Am fixing it as well.

**Comment by nickmarden** [comment]
@viclim I noticed one JS-related issue in the active_scaffold-related pages of the super-admin area:

screen_shot_2018-03-28_at_2_58_31_pm

Could you take a look?

File: Gemfile (hunk 2 above)
Comment (nickmarden) [reply Gemfile]: @viclim could you please use https://github.com/GSA/jquery-rails/commit/c0c56208f57746a4bc63431d5e771cea4486efca instead? It includes your changes but lives in a GSA-owned fork which is probably better in terms of maintainability.

File: app/assets/stylesheets/legacy_sass/sayt/custom.sass [hunk 3]
```diff
@@ -340,6 +340,13 @@
     text-align: left
     font-weight: bold
     font-size: smaller
+    /* jQuery UI 1.12 toggle .ui-state-active instead of .ui-state-hover when hovering autocomplete selection
```
Comment (nickmarden) [inline app/assets/stylesheets/legacy_sass/sayt/custom.sass]: @viclim can you confirm that an expected result of this change would be the following HTML differences?

new_type_ahead
old_type_ahead

**Comment by nickmarden** [comment]
@viclim just a gentle reminder to please update your Gemfile to point to https://github.com/GSA/jquery-rails/commit/c0c56208f57746a4bc63431d5e771cea4486efca

**Comment by nickmarden** [comment]
Closing in favor of #20

**Pull request closed without merging**